packages/block-library/src/navigation/edit/index.js

```diff
@@ -658,29 +693,6 @@ function Navigation( {
 					</InspectorControls>
 				) }
 				<nav { ...blockProps }>
-					{ isPlaceholderShown && (
```

Comment:
These complex loading states have been extracted to separate conditionals each of which returns early with a dedicated render.